src/core/createConsent.js

```diff
@@ -0,0 +1,273 @@
+      }
+
+      consentState.setOptInPreferences(preferencesByPurpose);
+
+      const payload = createConsentRequestPayload();
+      return lifecycle
+        .onBeforeConsentRequest({
+          payload
+        })
+        .then(() => {
+          payload.setPurposes(preferencesByPurpose);
+          return sendEdgeNetworkRequest({
+            payload,
+            action: "opt-in"
+          });
+        })
+        .then(() => {
+          // We optimistically use the preferences the user provided right away,
+          // but if the server doesn't agree and sends back a cookie value that
+          // conflicts, we adopt whatever the server said since that ultimately
+          // is what will be used if the user refreshes the page.
+          consentState.refreshFromCookies();
+        })
+        .catch(error => {
+          consentState.refreshFromCookies();
+          throw error;
+        });
+    },
+    /**
+     * Update the purposes the user has opted out of. Only to be called by the
+     * Privacy component.
+     * @param {Object} preferencesByPurpose An object where the key is
+     * the purposes name and the value is a boolean indicating whether the user
+     * has opted out of the purpose.
+     */
+    setOptOutPurposes(preferencesByPurpose) {
+      consentState.setOptOutPreferences(preferencesByPurpose);
+
+      const payload = createConsentRequestPayload();
+      return lifecycle
+        .onBeforeConsentRequest({
+          payload
+        })
+        .then(() => {
+          payload.setPurposes(preferencesByPurpose);
+          return sendEdgeNetworkRequest({
+            payload,
+            action: "opt-out"
+          });
+        })
+        .then(() => {
+          // We optimistically use the preferences the user provided right away,
+          // but if the server doesn't agree and sends back a cookie value that
+          // conflicts, we adopt whatever the server said since that ultimately
+          // is what will be used if the user refreshes the page.
+          consentState.refreshFromCookies();
+        })
+        .catch(error => {
+          consentState.refreshFromCookies();
+          throw error;
+        });
+    },
+    /**
+     * Returns a promise that is resolved once the user consents to all
+     * purposes. If the user has already consented to all purposes, the
+     * promise will already be resolved. If the user consents to no purposes,
+     * the promise will be rejected.
+     */
+    // TODO Once we support consenting to specific purposes, this
+    // method will accept an array of purpose names as an argument and
+    // will return a promise that will be resolved once the user has consented
+    // to the specified purposes.
+    whenConsented() {
+      const deferred = defer();
+      deferredsAwaitingConsent.push(deferred);
+      processConsentStatus(consentState.getStatus());
+      return deferred.promise;
+    }
+  };
+};
```

Comment:
There is a lot going on in this file.  Breaking these top-level functions into separate files under a `consent` folder may make unit testing easier.

Reply:
Yeah I think so too. I'll do that--probably in a different PR to make the changes more digestible.